### PR TITLE
feat(search): embedded view dropdown

### PR DESCRIPTION
**Pull request by joshuaeilers**
Home
Screenshot 2023-08-08 at 4 03 38 PM

Home autocomplete open:
Screenshot 2023-08-08 at 4 03 53 PM

Home views dropdown open:
Screenshot 2023-08-08 at 4 03 43 PM

Search:
Screenshot 2023-08-08 at 4 03 32 PM

Domains:
Screenshot 2023-08-08 at 4 04 04 PM

### CODE REVIEW

File: smoke-test/tests/cypress/cypress/e2e/views/view_select.js [hunk 1]
```diff
@@ -1,86 +1,88 @@
...
+      if (err.message.includes(resizeObserverLoopErrRe)) {
+        return false;
+      }
     });
+
+    cy.goToStarSearchList();
+
+    cy.log("Create a View from the select");
+    cy.multiSelect("view-select", "Create new view");
+    cy.waitTextVisible("Create new view");
```
Comment (joshuaeilers) [inline smoke-test/tests/cypress/cypress/e2e/views/view_select.js:27]: This text changed, rest is prettier auto-formatting.

**Review by chriscollins3456 — APPROVED** [review]
this is looking great! I pulled your branch down and things look good locally to me as well.

One thing I totally forgot about but it's been around some time now (ever since we upgraded ant design) is that when you wrap dropdowns in tooltips or something like that, it always shows a hover state for each dropdown item like:
image

Not saying it needs to go in this PR, but if we're looking to upgrade our Views experience it might be nice to clean that up as well

File: datahub-web-react/src/app/entity/view/select/ViewSelect.tsx [hunk 2]
```diff
@@ -1,37 +1,72 @@
...
+                position: absolute;
+            }
+            &.ant-select-open .ant-select-selection-item {
+                color: ${(props) => props.theme.styles['primary-color']};
+            }
+            &:not(.ant-select-open) .ant-select-selection-item {
+                color: ${ANTD_GRAY_V2[10]};
+            }
+            .ant-select-selection-placeholder,
+            .ant-select-selection-item {
```
[the remark this answers is not in the document]
Comment (joshuaeilers) [reply datahub-web-react/src/app/entity/view/select/ViewSelect.tsx:62]: Yeah.... and found I can clean some of this up on my next commit.

File: datahub-web-react/src/app/entity/view/select/ViewSelect.tsx [hunk 3]
```diff
@@ -153,54 +189,62 @@ export const ViewSelect = () => {
         (publicViews.filter((view) => view.urn === selectedUrn)?.length || 0) > 0 ||
         false;
 
+    const handleDropdownVisibleChange = (isNowOpen: boolean) => {
+        setIsOpen(isNowOpen);
+    };
+
     return (
-        <>
-            <ViewSelectToolTip visible={selectedUrn === undefined}>
```
[the remark this answers is not in the document]
Comment (joshuaeilers) [reply datahub-web-react/src/app/entity/view/select/ViewSelect.tsx:158]: It looked kinda rough to me having a tooltip just popup and cover the search bar, so preference here was cleanliness. I could be wrong on this, happy to try to revert if you feel it should stay.

File: datahub-web-react/src/app/entity/shared/constants.ts [hunk 4]
```diff
@@ -20,6 +20,13 @@ export const ANTD_GRAY = {
     9: '#434343',
 };
 
+export const ANTD_GRAY_V2 = {
```
[the remark this answers is not in the document]
Comment (joshuaeilers) [reply datahub-web-react/src/app/entity/shared/constants.ts:23]: Pushed up a fix for the hover issue. It's due to nested dropdowns. You hover a parent item and it cascades the active state of the dom node down to the child dropdown highlighting all children.

**Pull request merged**